packages/now-next/src/index.ts

```diff
@@ -136,16 +142,29 @@ async function writeNpmRc(workPath: string, token: string) {
   );
 }
 
-function getNextVersion(packageJson: {
-  dependencies?: { [key: string]: string };
-  devDependencies?: { [key: string]: string };
-}) {
+async function getNextVersion(entryPath: string) {
   let nextVersion;
-  if (packageJson.dependencies && packageJson.dependencies.next) {
-    nextVersion = packageJson.dependencies.next;
-  } else if (packageJson.devDependencies && packageJson.devDependencies.next) {
-    nextVersion = packageJson.devDependencies.next;
+
+  try {
+    // First try to resolve the `next` dependency and get the real version from its
+    // package.json. This allows the builder to be used with frameworks like Blitz that
+    // bundle Next but where Next isn't in the project root's package.json
+    nextVersion = require(resolveFrom(entryPath, 'next/package.json')).version;
```

Comment:
When this function gets called, `npm install` will not have been executed, you'll need to move the `runNpmInstall` further up

Reply:
I just pushed a fix for this.

The legacy next handling modifies the package.json, so npmInstall can't be moved before the version check. And the version is required to know if it's legacy or not.

So instead of moving npmInstall, I'm simply running `getNextVersion()` a second time after npmInstall.